sda-download/api/api.go

```diff
@@ -9,18 +9,24 @@ import (
 	"time"
 
 	"github.com/gin-gonic/gin"
+	"github.com/neicnordic/sda-download/api/middleware"
 	"github.com/neicnordic/sda-download/api/s3"
 	"github.com/neicnordic/sda-download/api/sda"
 	"github.com/neicnordic/sda-download/internal/config"
 	log "github.com/sirupsen/logrus"
 )
 
-// SelectedMiddleware is used to control authentication and authorization
-// behaviour with config app.middleware
-// available middlewares:
-// "default" for TokenMiddleware
-var SelectedMiddleware = func() gin.HandlerFunc {
-	return nil
+// SelectedMiddleware returns the middleware chain based on configuration.
+// For example, config.Config.App.Middleware could be "default", "token", etc.
+var SelectedMiddleware = func() []gin.HandlerFunc {
+	switch strings.ToLower(config.Config.App.Middleware) {
+	case "default":
+		return middleware.ChainDefaultMiddleware()
```

Comment:
```suggestion
		return []gin.HandlerFunc{
		middleware.TokenMiddleware(),
		middleware.ClientVersionMiddleware(),
	}
```

There is not need to hide this behind a separate function call.